src/rpdk/core/contract/resource_client.py

```diff
@@ -290,6 +290,18 @@ def is_primary_identifier_equal(
             for primary_identifier in primary_identifier_path
         )
 
+    @staticmethod
```

Comment:
make this an object method and use self.write_only_properties instead in the inside. it looks like everywhere it is used it uses that